File: c++/encrypt.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <streambuf>
#include <stdio.h>
#include <stdlib.h>

using namespace std;

void encrypt(char* infile, char* outfile, char* key){
	ifstream in(infile);
	if(in)
	{
		string plaintext = string(istreambuf_iterator<char>(in), istreambuf_iterator<char>());
		string ciphertext = "";
		for (int i = 0; i < plaintext.length(); i++)
		{	
			int k_ind = 0;
			if(k_ind == sizeof(key)) k_ind = 0;
			ciphertext += char((int(plaintext[i]) + int(key[k_ind]))%255);
			k_ind++;
		}
		ofstream out(outfile);
		out << ciphertext;
		out.close();
	}
	else
	{
		cout << "Input file does not exist\n";
		exit(EXIT_FAILURE);
	}
	in.close();
	return;
}

int main(int argc, char** argv){
	if(argc != 4){
		cout << "Invalid number of arguments\n";
		exit(EXIT_FAILURE);
	}
	encrypt(argv[1], argv[2], argv[3]);
	return(0);
}

File: python/encrypt.py
import sys

def return_args():
	if len(sys.argv) != 4:
		sys.exit("Invalid number of arguments")
	return sys.argv

def encrypt(infile, outfile, key):
	try:
		infile = open(infile, "r")
		outfile = open(outfile, "w")
		ciphertext = ""
		k_ind = 0
		for char in infile.read():
			if k_ind == len(key):
				k_ind = 0
			ciphertext += chr((ord(char) + ord(key[k_ind]))%255)
			k_ind += 1
		outfile.write(ciphertext)
		infile.close()
		outfile.close()	
	except IOError:
		sys.exit("Input file does not exist")

def main():
	args = return_args()
	encrypt(args[1], args[2], args[3])

main()
